homeassistant/components/poolstation/sensor.py

```diff
@@ -0,0 +1,110 @@
+"""Support for Poolstation sensors."""
+from __future__ import annotations
+
+from pypoolstation import Pool
+
+from homeassistant.components.sensor import SensorEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import DEVICE_CLASS_TEMPERATURE, PERCENTAGE, TEMP_CELSIUS
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import PoolstationDataUpdateCoordinator
+from .const import COORDINATORS, DEVICES, DOMAIN
+from .entity import PoolEntity
+
+PH_SUFFIX = " pH"
+TEMPERATURE_SUFFIX = " Temperature"
+SALT_SUFFIX = " Salt concentration"
+ELECTROLYSIS_SUFFIX = " Electrolysis"
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the poolstation sensors."""
+    stations = hass.data[DOMAIN][config_entry.entry_id][DEVICES]
+    coordinators = hass.data[DOMAIN][config_entry.entry_id][COORDINATORS]
+    entities: list[PoolEntity] = []
+    for pool_id, pool in stations.items():
+        coordinator = coordinators[pool_id]
+        entities.append(PoolPhSensor(pool, coordinator))
+        entities.append(PoolTemperatureSensor(pool, coordinator))
+        entities.append(PoolSaltConcentrationSensor(pool, coordinator))
+        entities.append(PoolElectrolysisSensor(pool, coordinator))
+
+    async_add_entities(entities)
+
+
+class PoolPhSensor(PoolEntity, SensorEntity):
```

Comment:
Considering the sensors are pretty much alike, maybe just have a single `PoolSensor` that takes in the necessary variables (name, attribute name, icon, unit)?

Reply:
I'm not convinced it's worth creating an abstraction that will make the code harder to reason about just to save a few lines of code. I'd rather have the code be as silly a possible in this case.

Reply:
Just a suggestion, it would make adding new sensors (in case needed at some point) easier. Your call!